### ISSUE TITLE
Trusted App "App Compatibility Helper" for collecting domains that cause HTTPS connection errors

### ISSUE THREAD

**Issue by mainzelM** [eblocker/eblocker#39]
Create a new TrustedApp that automatically collects all domains that were noticed to produce HTTPS conections errors.

Basic functionality:
* eBlocker provides a new built-in Trusted App named "App Compatibility Helper", with an initially empty exception list. It is initially disabled.
* The user can enable the Trusted App like any other
* If recording of SSL connection errors is enabled, eBlocker automatically adds the domains that cause SSL connection errors to the exception list of the new Trusted App. To prevent domains that produce SSL connection errors because of a browser/network problem, a domain must occur twice in the list to be be added.
* The user can manually add new domains to the Trusted App, even though this is not an intended use-case
* The user can manually remove domains from the Trusted App but as long as the failure for the domain is kept in eblocker's memory (i.e. for 7 days), it will be re-added automatically (upon the next HTTPS error). (Does not work at all currently)
* If the user clears the list of HTTPS connection errors, the new Trusted App will be cleared as well (as a lightweight mechanism to clear the App)
* Does not add domains that are blocked by an active rule
* If the new Trusted App is not enabled it will automatically appear within the "Suggested Trusted Apps" section. This is probably a nice side-effect and can be taken as is.
* If the new Trusted App is not enabled, it will still collect all HTTPS error domains. Therefore, after implementing this feature, the list in HTTPS->Communication Failures->Suggested Trusted Websites would always be empty (well, except for the domains that we don't add automatically because they are blocked). To avoid this, we add special handling for the new Trusted App in the suggestions so that the existing suggestions appear as before

Open questions:

* Due to the automatic additions, the list will be marked as modified soon. A future update of the list definition will yield the following message "System updates have not been applied to this exception list due to custom modifications. Please reset this exception list to get the latest updates. You may make a copy of your changes to apply them again if required."

Possible extensions:
* Allow users to permanently delete domains from the Trusted App, i.e. the domain will not be re-added if it produces a SSL error in the future (or is still in eBlockers memory)
    * Pro: Certainly a nice feature for advanced users, but beginners won't care much
    * Cons: More complexity in the code (must catch delete, must store deleted domains)
    * Idea:
        * Add a new type (format) of Blocker Lists that contains a list of domains or patterns which can be edited by the user
        * Using this new type, add an "Always distrust" list. 
        * If the user deletes a domain from the Trusted App, she can choose to "always distrust" the domain, in which case it will be added to the "Always distrust" list. As long as it is there, it is never automatically re-added to the Trusted App.
* Do not add domains that are already covered by other Trusted Apps
    * Pro: Users can simply enable the existing Trusted App, so prevent duplication
    * Cons: Still requires interaction for some domains whereas others magically/automatically work out of the box. This may be counter-intuitive.
* Do not add sub-domains if the domain is already present, remove sub-domains if domain itself is added
    * Pro: Possibly keeps the list smaller
    * Cons: Should be easy to implement but does not fit nicely into the existing framework as the user-defined Trusted Apps don't have such a mechanism
* Add hints about the new Trusted App to other parts of the eBlocker settings
